apps/typeform/frontend/src/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import { AppExtensionSDK } from 'contentful-ui-extensions-sdk';
+import get from 'lodash/get';
+import {
+  Heading,
+  Paragraph,
+  Typography,
+  TextLink,
+  Select,
+  Option,
+  FormLabel
+} from '@contentful/forma-36-react-components';
+import FieldSelector from './FieldSelector';
+import {
+  CompatibleFields,
+  ContentType,
+  Hash,
+  EditorInterface,
+  InstallationParameters,
+  SelectedFields,
+  WorkspaceOption,
+  WorkspacesResponse
+} from '../typings';
+import {
+  getCompatibleFields,
+  editorInterfacesToSelectedFields,
+  selectedFieldsToTargetState,
+  validateParameters
+} from '../utils';
+
+import { styles } from './styles';
+
+// @ts-ignore 2307
+import logo from './config-screen-logo.svg';
+
+interface Props {
+  sdk: AppExtensionSDK;
+  expireSoon: boolean;
+}
+
+interface State {
+  workspaceId: string;
+  accessToken: string;
+  workspaces: WorkspaceOption[];
+  contentTypes: ContentType[];
+  selectedContentTypes: string[];
+  selectedFields: SelectedFields;
+  compatibleFields: CompatibleFields;
+}
+
+export class AppConfig extends React.Component<Props, State> {
+  state: State = {
+    contentTypes: [],
+    compatibleFields: {},
+    workspaces: [],
+    selectedContentTypes: [],
+    selectedFields: {},
+    workspaceId: '',
+    accessToken: (window.localStorage.getItem('token') as string) || ''
+  };
+
+  async componentWillMount() {
```

Comment:
I think we should use the did mount method.

```suggestion
  async componentDidMount() {
```